### PR TITLE
Fix #306: Implement approval UI extension payload

### LINKED ISSUE #306: Implement approval UI extension payload
We should add data package for additional UI hints:

```json
"ui": {
  "flipButtons": true,
  "blockApprovalOnCall": true,
  "preApprovalScreen": {
    "type": "WARNING",
    "heading": "Dejte si pozor!",
    "message": null,
    "items": [
      "Aktivujete novou aplikaci a povolujete přístup k Vašim účtům",
      "Ujistěte se, že aktivace probíhá na Vašem zařízení",
      "Pokud jste byli vyzvání k této operaci v souvislosti s platbou, odmítněte ji"
    ],
    "approvalType": "SLIDER"
  }
}
```

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

**Review by romanstrobl — APPROVED** [review]
Looks good.

**Review by kober32 — APPROVED** [review]
OK with me

**Comment by kober32** [comment]
Just to bear it in mind. The offline operation needs a similar change
- flag for buttons flip (`X`?)
- flag for fraud warning (`F`?)
- flag for disabling authorization when on call (`C`?)

File: docs/sql/oracle/create-schema.sql [hunk 1]
```diff
@@ -25,7 +25,8 @@ CREATE TABLE ES_OPERATION_TEMPLATE (
     LANGUAGE VARCHAR2(8 CHAR) NOT NULL,
     TITLE VARCHAR2(255 CHAR) NOT NULL,
     MESSAGE BLOB NOT NULL,
-    ATTRIBUTES BLOB
+    ATTRIBUTES BLOB,
+    UI BLOB
```
Comment (banterCZ) [inline docs/sql/oracle/create-schema.sql:29]:
```suggestion
    ATTRIBUTES CLOB,
    UI CLOB
```

**Review by banterCZ — CHANGES_REQUESTED** [review]
Please change the column type to `CLOB`.

**Review by banterCZ — APPROVED** [review]

File: enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+        assertNotNull(result.getUi());
+
+        final UiExtensions ui = result.getUi();
+        assertEquals(true, ui.getFlipButtons());
+        assertEquals(true, ui.getBlockApprovalOnCall());
+        assertNotNull(ui.getPreApprovalScreen());
+
+        final PreApprovalScreen preApprovalScreen = ui.getPreApprovalScreen();
+        assertEquals(PreApprovalScreen.ScreenType.WARNING, preApprovalScreen.getType());
```
Comment (banterCZ) [inline enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java:158]: @kober32 Just highlight, that in this case, you will get 
```json
{
  "flipButtons": true,
  "blockApprovalOnCall": true,
  "preApprovalScreen": {
    "type": "WARNING"
  }
}
```

File: enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+
+        final Operation result = tested.convert(operationDetail, operationTemplate);
+
+        assertNull(result.getUi());
+    }
+
+    @Test
+    void testConvertUi_overriddenByEnrollment() throws Exception {
```
Comment (romanstrobl) [inline enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java:57]: This looks really ugly, is it necessary?

File: enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        final OperationDetailResponse operationDetail = new OperationDetailResponse();
+        operationDetail.setSignatureType(List.of(SignatureType.KNOWLEDGE));
+        operationDetail.setStatus(OperationStatus.APPROVED);
+        operationDetail.setRiskFlags("C");
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+        operationTemplate.setUi("{\n" +
```
Comment (romanstrobl) [inline enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java:64]: We do not want to use Czech language in upstream. The String should be moved into a constant.

**Review by romanstrobl — CHANGES_REQUESTED** [review]
Please avoid using Czech language in upstream.

File: enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java (hunk 2 above)
Comment (kober32) [reply enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java:158]: OK, just note that with this, the current app will fail to display the operation as other things in `preApprovalScreen` are required, so this needs changes in the app - I'm creating an issue

File: enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java (hunk 3 above)
Comment (banterCZ) [reply enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java:57]: What is ugly, the underscore? I was inspired by Spock but I will remove it.

File: enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java (hunk 4 above)
Comment (banterCZ) [reply enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java:64]: Understood, I will change it. I took the template from the issue, where it is in Czech.

**Review by romanstrobl — APPROVED** [review]
I think we can merge this one now. The changes in mobile app for risk warning are pending.

**Pull request merged**